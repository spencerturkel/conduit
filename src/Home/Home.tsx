import * as React from 'react';
import {ComponentType, StatelessComponent} from 'react';

import setDisplayName from '../set-display-name';

const Home = (PopularTags: ComponentType, ArticleListing: ComponentType): StatelessComponent => {
    const sfc: StatelessComponent = () => (
        <div className="home-page">
            <div className="banner">
                <div className="container">
                    <h1 className="logo-font">conduit</h1>
                    <p>A place to share your knowledge.</p>
                </div>
            </div>

            <div className="container page">
                <div className="row">
                    <div className="col-md-9">
                        <ArticleListing />
                    </div>

                    <div className="col-md-3">
                        <PopularTags />
                    </div>
                </div>
            </div>
        </div>
    );

    setDisplayName(sfc, Home, PopularTags, ArticleListing);

    return sfc;
};

export default Home;
