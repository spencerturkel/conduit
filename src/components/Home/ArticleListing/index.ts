export {default} from './ArticleListing';
export * from './ArticleListing';
