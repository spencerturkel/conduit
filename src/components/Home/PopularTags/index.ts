export {default} from './PopularTags';
export * from './PopularTags';
