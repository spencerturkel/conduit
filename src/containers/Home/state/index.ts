import * as _tags from './tags';
import {createSelector} from 'reselect';

export interface State {
    tags: _tags.State;
}

export const reducers = {
    tags: _tags.reducer,
};

const selectTags = (state: State): _tags.State => state.tags;

export const tags = {
    fetchAll: _tags.fetchAll,
    fetchAllError: _tags.fetchAllError,
    loadAll: _tags.loadAll,
    fetchAll$: _tags.fetchAll$,
    selectError: createSelector(selectTags, _tags.selectError),
    selectLoading: createSelector(selectTags, _tags.selectLoading),
    selectTags: createSelector(selectTags, _tags.selectTags),
};
