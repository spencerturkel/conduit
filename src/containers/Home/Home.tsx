import * as React from 'react';
import {ComponentClass} from 'react';
import {Link as ReactRouterLink} from 'react-router-dom';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';

import Home from '../../components/Home';
import ArticleListing, {ArticlePreview} from '../../components/Home/ArticleListing';
import FeedPicker from '../../components/Home/ArticleListing/FeedPicker';
import PopularTags from '../../components/Home/PopularTags';
import {LinkProps} from '../../components/link-props';
import * as state from '../../state';

const AppArticleListing = (() => {
    const articlePreviews: ArticlePreview[] = [
        {
            author: {
                image: 'https://i.stack.imgur.com/xHWG8.jpg',
                username: 'jake',
            },
            createdAt: new Date('2017-10-12'),
            description: 'This is the description for the post.',
            favorited: true,
            favoritesCount: 27,
            slug: 'test-slug',
            title: 'How to build webapps that scale',
        },
        {
            slug: 'how-to-train-your-dragon-2',
            title: 'How to train your dragon 2',
            description: 'So toothless',
            createdAt: new Date('2016-02-18T03:22:56.637Z'),
            favorited: false,
            favoritesCount: 0,
            author: {
                username: 'jake',
                image: 'https://i.stack.imgur.com/xHWG8.jpg',
            },
        },
    ];

    const AppFeedPicker = () => <FeedPicker activeFeed={'global'} loggedIn={false} />;

    const Result = ArticleListing(AppFeedPicker, ReactRouterLink as ComponentClass<LinkProps>);

    return () => <Result previews={articlePreviews} onFavoriteCountClick={slug => alert(`Favorited slug ${slug}`)} />;
})();

const AppPopularTags = (() => {
    const Result = connect(createSelector(state.tags.selectTags, allTags => ({tags: allTags})))(PopularTags());

    return () => <Result onTagClicked={tag => alert('clicked ' + tag)} />;
})();

const HomeContainer = Home(AppPopularTags, AppArticleListing);

export default HomeContainer;
