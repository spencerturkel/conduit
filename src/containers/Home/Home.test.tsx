import {shallow} from 'enzyme';
import * as React from 'react';

import Home from './Home';

it('matches the snapshot', () => {
    const sut = shallow(<Home />);
    expect(sut).toMatchSnapshot();
});
