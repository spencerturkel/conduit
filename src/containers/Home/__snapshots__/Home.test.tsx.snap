// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home(, ) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="banner"
>
          <div
                    className="container"
          >
                    <h1
                              className="logo-font"
                    >
                              conduit
                    </h1>
                    <p>
                              A place to share your knowledge.
                    </p>
          </div>
</div>,
        <div
          className="container page"
>
          <div
                    className="row"
          >
                    <div
                              className="col-md-9"
                    >
                              <Unknown />
                    </div>
                    <div
                              className="col-md-3"
                    >
                              <Unknown />
                    </div>
          </div>
</div>,
      ],
      "className": "home-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <h1
                        className="logo-font"
            >
                        conduit
            </h1>
            <p>
                        A place to share your knowledge.
            </p>
</div>,
          "className": "banner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="logo-font"
>
                conduit
</h1>,
              <p>
                A place to share your knowledge.
</p>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "conduit",
                "className": "logo-font",
              },
              "ref": null,
              "rendered": "conduit",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A place to share your knowledge.",
              },
              "ref": null,
              "rendered": "A place to share your knowledge.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-md-9"
            >
                        <Unknown />
            </div>
            <div
                        className="col-md-3"
            >
                        <Unknown />
            </div>
</div>,
          "className": "container page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-md-9"
>
                <Unknown />
</div>,
              <div
                className="col-md-3"
>
                <Unknown />
</div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown />,
                "className": "col-md-9",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown />,
                "className": "col-md-3",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="banner"
>
            <div
                        className="container"
            >
                        <h1
                                    className="logo-font"
                        >
                                    conduit
                        </h1>
                        <p>
                                    A place to share your knowledge.
                        </p>
            </div>
</div>,
          <div
            className="container page"
>
            <div
                        className="row"
            >
                        <div
                                    className="col-md-9"
                        >
                                    <Unknown />
                        </div>
                        <div
                                    className="col-md-3"
                        >
                                    <Unknown />
                        </div>
            </div>
</div>,
        ],
        "className": "home-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <h1
                            className="logo-font"
              >
                            conduit
              </h1>
              <p>
                            A place to share your knowledge.
              </p>
</div>,
            "className": "banner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="logo-font"
>
                  conduit
</h1>,
                <p>
                  A place to share your knowledge.
</p>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "conduit",
                  "className": "logo-font",
                },
                "ref": null,
                "rendered": "conduit",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A place to share your knowledge.",
                },
                "ref": null,
                "rendered": "A place to share your knowledge.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="col-md-9"
              >
                            <Unknown />
              </div>
              <div
                            className="col-md-3"
              >
                            <Unknown />
              </div>
</div>,
            "className": "container page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-md-9"
>
                  <Unknown />
</div>,
                <div
                  className="col-md-3"
>
                  <Unknown />
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Unknown />,
                  "className": "col-md-9",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Unknown />,
                  "className": "col-md-3",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
